conjure-go-contract/codecs/snappy.go

```diff
@@ -68,20 +67,16 @@ func (c codecSNAPPY) Encode(w io.Writer, v interface{}) error {
 	if err != nil {
 		return err
 	}
-
-	encoded, err := c.Marshal(data)
-	if err != nil {
-		return err
-	}
+	encoded := snappy.Encode(nil, data)
 	_, err = w.Write(encoded)
 	return err
 }
 
 func (c codecSNAPPY) Marshal(v interface{}) ([]byte, error) {
-	data, ok := v.([]byte)
-	if !ok {
-		return nil, werror.Error("failed to compress data from type which is not of type []byte")
```

Comment:
this was always incorrect -- Marshal should have been calling the contentCodec rather than expect its input to already be encoded.